bin/sozo/src/commands/auth.rs

```diff
@@ -15,17 +19,131 @@ pub struct AuthArgs {
     pub command: AuthCommand,
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct ModelContract {
+    pub model: FieldElement,
+    pub contract: FieldElement,
+}
+
+impl FromStr for ModelContract {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parts: Vec<&str> = s.split(',').collect();
+
+        let (model, contract_part) = match parts.as_slice() {
+            [model, contract] => (model.to_string(), *contract),
+            _ => anyhow::bail!(
+                "Model and contract address are expected to be comma separated: `sozo auth writer \
+                 model_name,0x1234`"
+            ),
+        };
+
+        let model = cairo_utils::str_to_felt(&model)
+            .map_err(|_| anyhow::anyhow!("Invalid model name: {}", model))?;
+
+        let contract = FieldElement::from_hex_be(contract_part)
+            .map_err(|_| anyhow::anyhow!("Invalid contract address: {}", contract_part))?;
+
+        Ok(ModelContract { model, contract })
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum ResourceType {
+    Contract(String),
+    Model(FieldElement),
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct OwnerResource {
+    pub owner: FieldElement,
+    pub resource: ResourceType,
+}
+
+impl FromStr for OwnerResource {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parts: Vec<&str> = s.split(',').collect();
+
+        let (owner_part, resource_part) = match parts.as_slice() {
+            [owner, resource] => (*owner, *resource),
+            _ => anyhow::bail!(
+                "Owner and resource are expected to be comma separated: `sozo auth owner \
+                 0x1234,resource_type:resource_name`"
+            ),
+        };
+
+        let owner = FieldElement::from_hex_be(owner_part)
+            .map_err(|_| anyhow::anyhow!("Invalid owner address: {}", owner_part))?;
+
+        let resource_parts: Vec<&str> = resource_part.split(':').collect();
```

Comment:
If the resource can be `contract:contract_name`, we're expecting the fully qualifier name of the contract or the short name? It will fail here if we have to pass the fully qualified name due to `:` split.

Reply:
Agree, this would be an issue for fully qualified name. 

We can use [`split_once`](https://doc.rust-lang.org/std/primitive.str.html#method.split_once) for this to capture the first occurrence of `:` if we wanna maintain this format. 

Considering most contract name would be fully qualified name, i think this format can cause a bit of confusion when user do something like `contract:examples::action::action_1`

Reply:
hm yeah good point. i was following the starkli schema pattern but perhaps we should do something difference

Reply:
thinking about this a bit more, i feel like it is ok. so going to plan to keep as is, unless you guys feel strongly about it

Reply:
Not really, this is a needed feature so don't wanna block too much. We can always address it if it becomes an actual concern later.